osu.Game/Overlays/Music/PlaylistList.cs

```diff
@@ -103,6 +111,50 @@ public ItemSearchContainer()
                 LayoutDuration = 200;
                 LayoutEasing = Easing.OutQuint;
             }
+
+            public Action<List<BeatmapSetInfo>> UpdateBeatmapSets;
+            private List<BeatmapSetInfo> beatmapSets = new List<BeatmapSetInfo>();
+
+            private Vector2 mousePosition;
+
+            protected override bool OnMouseMove(InputState state)
+            {
+                mousePosition = state.Mouse.Position;
+                return base.OnMouseMove(state);
+            }
+
+            protected override void Update()
+            {
+                base.Update();
+                PlaylistItem item = Children.FirstOrDefault(i => i.DragStartOffset != Vector2.Zero);
+                if (item == default(PlaylistItem))
+                    return;
+                List<PlaylistItem> childrenList = Children.ToList();
+                item.MoveTo(new Vector2(0, (mousePosition - item.DragStartOffset).Y));
+
+                int targetIndex = Children.Count(i => i.Position.Y - item.Size.Y / 2 < item.Position.Y - i.Size.Y / 2);

```

Comment:
Also linear stuff reeeeeally doesn't scale to 100k+ beatmaps. Might be a good idea to just use a `SortedList` and then use `IndexOf`.